tools/check_format.py

```diff
@@ -409,12 +419,19 @@ def checkFileContents(file_path, checker):
     # notes have a different format.
     checkCurrentReleaseNotes(file_path, error_messages)
 
+  format_flag = True
   for line_number, line in enumerate(readLines(file_path)):
 
+    if line.find("// clang-format on"):
+      format_flag = True
+    if line.find("// clang-format off"):
+      format_flag = False
+
     def reportError(message):
       error_messages.append("%s:%d: %s" % (file_path, line_number + 1, message))
 
-    checker(line, file_path, reportError)
+    if format_flag:
+      checker(line, file_path, reportError)
```

Comment:
These changes may require an independent review, and to be force pushed because it looks like the state of tools/check_format.py's check_format_test itself may be currently broken on master.

Reply:
Our abuse of list().find() in python, no error on master. Fixed and resolving review